Gateway/Request/CheckoutDataBuilder.php

```diff
@@ -25,188 +25,236 @@
-				'amountExcludingTax' => $itemAmount,
-				'taxAmount' => $itemVatAmount,
-				'description' => $description,
-				'quantity' => $numberOfItems,
-				'taxCategory' => 'None',
-				'taxPercentage' => $itemVatPercentage
-			];
-		}
-
-		// Shipping cost
-		if ($this->quote->getShippingAddress()->getShippingAmount() > 0 || $this->quote->getShippingAddress()->getShippingTaxAmount() > 0) {
-
-			$priceExcludingTax = $this->quote->getShippingAddress()->getShippingAmount() - $this->quote->getShippingAddress()->getShippingTaxAmount();
-
-			$taxClassId = $this->taxConfig->getShippingTaxClass($this->storeManager->getStore()->getId());
-
-			$formFields['lineItems'][] = [
-				'amountExcludingTax' => $priceExcludingTax,
-				'taxAmount' => $this->quote->getShippingAddress()->getShippingTaxAmount(),
-				'description' => $order->getShippingDescription(),
-				'quantity' => 1,
-				'taxPercentage' => $this->quote->getShippingAddress()->getShippingTaxAmount()
-			];
-		}
-
-		return $formFields;
-	}
+    /**
+     * @var \Adyen\Payment\Helper\Data
+     */
+    private $adyenHelper;
+
+    /**
+     * @var \Magento\Store\Model\StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @var \Magento\Checkout\Model\Session
+     */
+    private $checkoutSession;
+
+    /**
+     * @var \Magento\Quote\Model\Quote
+     */
+    private $quote;
+
+    /**
+     * @var \Magento\Tax\Model\Config
+     */
+    protected $taxConfig;
+
+    protected $adyenLogger;
+
+    /**
+     * CheckoutDataBuilder constructor.
+     *
+     * @param \Adyen\Payment\Helper\Data $adyenHelper
+     * @param \Magento\Store\Model\StoreManagerInterface $storeManager
+     * @param \Magento\Checkout\Model\Session $checkoutSession
+     * @param \Magento\Tax\Model\Config $taxConfig
+     */
+    public function __construct(
+        \Adyen\Payment\Helper\Data $adyenHelper,
+        \Magento\Store\Model\StoreManagerInterface $storeManager,
+        \Magento\Checkout\Model\Session $checkoutSession,
+        \Magento\Tax\Model\Config $taxConfig,
+        \Adyen\Payment\Logger\AdyenLogger $adyenLogger
+    ) {
+        $this->adyenHelper = $adyenHelper;
+        $this->storeManager = $storeManager;
+        $this->checkoutSession = $checkoutSession;
+        $this->quote = $checkoutSession->getQuote();
+        $this->taxConfig = $taxConfig;
+        $this->adyenLogger=$adyenLogger;
+
+    }
+
+    /**
+     * @param array $buildSubject
+     * @return mixed
+     */
+    public function build(array $buildSubject)
+    {
+        /** @var \Magento\Payment\Gateway\Data\PaymentDataObject $paymentDataObject */
+        $paymentDataObject = \Magento\Payment\Gateway\Helper\SubjectReader::readPayment($buildSubject);
+        $payment = $paymentDataObject->getPayment();
+        $order = $payment->getOrder();
+        $storeId = $order->getStoreId();
+        $request = [];
+
+        // do not send email
+        $order->setCanSendNewEmailFlag(false);
```

Comment:
for Boleto we want to send out email after order creation. THis will provide them the PDF so they can go to there bank.

Reply:
Now inside boleto check we send the email